docs/r3/remote_finality_guarantors.md

```diff
@@ -0,0 +1,67 @@
+
+Suppose that our two networks are a Corda network and an EVM network, and Alice@Corda has agreed to transfer an asset to Bob@Corda in exchange for Bob@EVM making a reciprocating transfer to Alice@EVM.
+
+Alice@Corda constructs a _draft transaction_ which would place the asset on the Corda network in a lock, from which it could be released to Bob@Corda by presenting proof of action of the reciprocating transfer from Bob@EVM to Alice@EVM.
+
+Bob@Corda verifies that the draft transaction will do what is agreed and that the lock conditions will be able to be satisfied by the agreed action of Bob@EVM. It then provides Bob@EVM with the hash of the draft transaction and the public key of the notary that will finalise it, and Bob@EVM sets up the lock on the other network using these values.
+
+If Alice@Corda now notarises the draft transaction, they will obtain the notary's signature on the transaction hash, which they can pass to Alice@EVM to authorise the transfer of the asset on the EVM network to themselves. This in turn will publicly broadcast the EVM transfer transaction, so that Bob@Corda can obtain the proof of action needed to unlock the asset on the Corda network.
+
+The complete proof of action on the Corda side consists of:
+
+1. Proof that a transaction with a given hash _will perform_ the required action. This is obtained through vicarious trust: Bob@Corda validates the draft transaction and passes the transaction hash on to Bob@EVM, who proceeds on the basis of trust in Bob@Corda.
+2. Proof that a transaction with the given hash _was finalised_. This is obtained through a combination of vicarious trust (Bob@Corda has supplied the public key of the notary whose signature means that the draft transaction has been finalised) and offline-verifiable cryptographic proof (the EVM contract checks that the provided transaction hash has been signed with this public key as a condition of completing the transfer transaction).
+
+The complete proof of action on the EVM side consists of:
+
+1. Proof that a transaction _has performed_ the required action. This is obtained by inspecting the events recorded within the transaction's data.
+2. Proof that the transaction _was finalised_. This comes in two parts:
+    * A Merkle Patricia inclusion proof that the hash of the transaction is included in a block.
+    * Proof that the block having that hash was finalised on the EVM blockchain.
+
+Only the last part of the EVM proof of action requires remote finality observation, and it does so only in the case that finality on the remote chain is not deterministically verifiable by presenting checking validator signatures on the block hash.
+
+
+## Expanding the role of attestation
```

Comment:
This may be so, but note that the entire following section suggests an _expanded_ role for an additional trusted party, and argues for the usefulness of having a guarantor even in the case where one isn't absolutely needed!

Reply:
that indeed may be so. Doesn't mean I have to _like_ it!

Reply:
See (from the new) Line 39 in this commit: https://github.com/hyperledger-labs/harmonia/pull/16/commits/e1c8f93bb7d2d1b32ac7a58d961adf2b06c7c62e